impeller/geometry/path_component.h

```diff
@@ -19,14 +19,65 @@ namespace impeller {
 ///        strip.
 class VertexWriter {
  public:
-  explicit VertexWriter(std::vector<Point>& points,
-                        std::vector<uint16_t>& indices);
+  virtual void EndContour() = 0;
 
-  ~VertexWriter() = default;
+  virtual void Write(Point point) = 0;
+};
+
+/// @brief A vertex writer that generates a triangle fan and requires primitive
+/// restart.
+class FanVertexWriter : public VertexWriter {
```

Comment:
This will only work for 16-bit index types. Perhaps you can take a void* index buffer and an IndexType argument to make this utility more generally usable. The restart marker you select can then depend on the index type passed in. Or you could template it.

This looks like a generally useful utility and its in the common geometry classes, so making it only work with 16 bit index types seems odd. And there are very guardrails against someone casting the wrong pointer type to uint16_t.

Reply:
We only use 16 bit index types though. If we need to make this configurable we can template-tize it if needed.